src/rooms/IPreRoomCreateHandler.ts

```diff
@@ -2,7 +2,14 @@ import { IRead } from '../accessors/index';
 import { IRoom } from './IRoom';
 
 export interface IPreRoomCreateHandler {
-    preRoomCreate(room: IRoom): IRoom;
+    /**
+     * Checks whether the handler actually shall execute.
+     * This method can basically implement a filter.
+     *
+     * @param room
+     * @param read
+     */
+    checkPreRoomCreate(room: IRoom, read: IRead): boolean;
 
-    isRoomApplicable(message: IRoom, read: IRead): boolean;
+    preRoomCreate(room: IRoom): IRoom;
```

Comment:
@graywolf336 I guess that this is not finished yet - just wanted to add I think that in such a method, a developer would do anything. This would reduce Rocketlet quality and the ability to expose the scope of a Rocketlet upfront.